src/Splat.Microsoft.Extensions.DependencyInjection/MicrosoftDependencyResolver.cs

```diff
@@ -0,0 +1,213 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Splat.Microsoft.Extensions.DependencyInjection
+{
+    /// <summary>
+    /// Microsoft DI implementation for <see cref="IDependencyResolver"/>.
+    /// </summary>
+    /// <seealso cref="IDependencyResolver" />
+    [SuppressMessage("Globalization", "CA1303", Justification = "Unnecessary warning")]
+    public class MicrosoftDependencyResolver : IDependencyResolver
+    {
+        private const string ImmutableExceptionMessage = "This container has already been built and cannot be modified.";
+        private readonly Lazy<IDictionary<string, IServiceScope>> _serviceScopes = new Lazy<IDictionary<string, IServiceScope>>(() => new Dictionary<string, IServiceScope>());
+        private IServiceCollection _serviceCollection;
+        private bool _isImmutable;
+        private IServiceProvider _serviceProvider;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MicrosoftDependencyResolver" /> class with an <see cref="IServiceCollection"/>.
+        /// </summary>
+        /// <param name="services">An instance of <see cref="IServiceCollection"/>.</param>
+        public MicrosoftDependencyResolver(IServiceCollection services = null)
+        {
+            _serviceCollection = services ?? new ServiceCollection();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MicrosoftDependencyResolver" /> class with a configured service Provider.
+        /// </summary>
+        /// <param name="serviceProvider">A ready to use service provider.</param>
+        public MicrosoftDependencyResolver(IServiceProvider serviceProvider)
+        {
+            if (serviceProvider == null)
+            {
+                throw new ArgumentNullException(nameof(serviceProvider));
+            }
+
+            _serviceProvider = serviceProvider;
+            _isImmutable = true;
+        }
+
+        /// <inheritdoc />
+        public virtual object GetService(Type serviceType, string contract = null)
+        {
+            if (serviceType == null)
+            {
+                throw new ArgumentNullException(nameof(serviceType));
+            }
+
+            var sp = GetServiceProvider(contract);
+            return sp.GetRequiredService(serviceType);
+        }
+
+        /// <inheritdoc />
+        public virtual IEnumerable<object> GetServices(Type serviceType, string contract = null)
+        {
+            if (serviceType == null)
+            {
+                throw new ArgumentNullException(nameof(serviceType));
```

Comment:
I believe that we should be failing gracefully and not throwing an exception in this case.  If memory serves there is a platform concern with registrations that this might cause problems.

Reply:
Why is it supposed to be null?

Reply:
I see the Splat resolver has a 'null' type value option. Will implement that.